README.md

```diff
@@ -1568,6 +1576,175 @@ In this example, the `NewUser` type implements the `IValidateable<NewUser>` inte
+        validation {
+            withValue s
+            disputeWithFact PasswordFailures.Empty isNotNull
+            disputeWithFact PasswordFailures.TooShort (fun (s: string) -> s.Length >= 8)
+            disputeWithFact PasswordFailures.TooLong (fun (s: string) -> s.Length <= 20)
+            qed Password
+        }
+        |> fromVCtx
+
+    let unwrap (Password s) = s
+
+type Username = private Username of string
+
+module Username =
+    let make s =
+        validation {
+            withValue s
+            disputeWithFact UsernameFailures.Empty isNotNull
+            qed Username
+        }
+        |> fromVCtx
+
+    let unwrap (Username s) = s
+```
+
+#### Step 3: Define the Model and Implement IValidateable
+
+Next, we define our `NewUserVM` model and implement the `IValidateable<'F>` interface, writing validations for the model using this library.
+In this case, the failure type of the `IValidateable<'F>` represented by `'F` is `NewUserFailures`.
+
+```fsharp
+[<CLIMutable>]
+type NewUserVM =
+    { Name: string option
+      Username: string option
+      Password: string option }
+
+    interface IValidateable<NewUserFailures> with
+        member this.Validate() =
+            validation {
+                // Validate the Name field
+                let! n =
+                    validation {
+                        withField (fun () -> this.Name)
+                        refuteWith (isRequired RequiredField)
+                        qed
+                    }
+
+                // Validate the Username field
+                and! un =
+                    validation {
+                        withField (fun () -> this.Username)
+                        refuteWith (isRequired RequiredField)
+                        refuteWithProof (Username.make >> Proof.mapInvalid InvalidUsername)
+                        qed
+                    }
+
+                // Validate the Password field
+                and! pw =
+                    validation {
+                        withField (fun () -> this.Password)
+                        refuteWith (isRequired RequiredField)
+                        refuteWithProof (Password.make >> Proof.mapInvalid InvalidPassword)
+                        qed
+                    }
+
+                return ()
+            }
+            |> fromVCtx
+```
+
+In this implementation, we use the `validation` computation expression to define the validation logic for each field.
+For the `Username` and `Password` fields, we use the `Username.make` and `Password.make` functions respectively to perform additional validation specific to these fields.
+
+#### Step 4: Define the Giraffe IModelValidation
+
+Now, we need to implement the Giraffe `IModelValidation<'T>` interface and its `Validate` method.
+This is what will apply the validations and throw an appropriate HTTP response.
+
+```fsharp
+[<CLIMutable>]
+type NewUserVM =
+    // ... nothing new here
+
+    interface IModelValidation<NewUserVM> with
```

Comment:
This implementation suggests validation without transformation. That's antithetical to the design of this library.